src/test/java/com/linecorp/armeria/client/manipulating/HttpHeaderManipulatingClientBuilderTest.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.manipulating;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.common.http.DefaultHttpRequest;
+import com.linecorp.armeria.common.http.HttpHeaders;
+import com.linecorp.armeria.common.http.HttpMethod;
+import com.linecorp.armeria.common.http.HttpRequest;
+
+import io.netty.util.AsciiString;
+
+
+public class HttpHeaderManipulatingClientBuilderTest {
+
+    @Test
+    public void testErrorOnNullOrEmptyHeader() {
+        HttpHeaderManipulatingClientBuilder builder = newBuilder();
+        boolean thrown = false;
+        try {
+            builder.add(null, "value");
+            fail("Exception not thrown on null header ");
+
+            builder.add(EMPTY_HEADER, "value");
+            fail("Exception not thrown on empty header");
+
+            builder.set(null, "value");
+            fail("Exception not thrown on null header");
+
+            builder.set(EMPTY_HEADER, "value");
+            fail("Exception not thrown on empty header");
+
+            builder.remove((AsciiString) null);
+            fail("Exception not thrown on null header");
+
+            builder.remove(EMPTY_HEADER);
+            fail("Exception not thrown on empty header");
+
+            builder.update(null);
+            fail("Exception not thrown on null updater");
+
+        } catch (Exception e) {
+            thrown = true;
+        }
+        assertEquals(true, thrown);
```

Comment:
Please use better ways to check exceptions. https://github.com/junit-team/junit4/wiki/exception-testing